core/src/main/scala/pureconfig/package.scala

```diff
@@ -278,8 +278,8 @@ package object pureconfig {
       files
         .map(parseFile)
         .foldLeft[Either[ConfigReaderFailures, Seq[TypesafeConfig]]](Right(Seq())) {
-          case (c1, c2) =>
-            ConfigConvert.combineResults(c1, c2)(_ :+ _)
+          case (c1, Left(ConfigReaderFailures(_: CannotReadFile, Nil))) => c1
```

Comment:
I cannot think of any case where a `CannotReadFile` can be accompanied by another failure, but logically speaking it makes sense to check all failures. I changed the behavior to what you suggested in ea17803c2f79049885867f4961ee054496630755.